magento/magento2
# Fixing bug with non-existent customer group breaking quote

### Description (*)

There's a bug when a customer group deleted, it's still referenced in active quotes which causes an error on the checkout page.
Such a quote becomes broken and the customer is not able to place an order. This pull request adds functionality for getting customer group from the current customer in case customer group saved to the quote does not exist.

### Manual testing scenarios (*)

1. Create a new customer group.
2. Create a new customer. Link that customer to the customer group created on the previous step.
3. Log in to the Magento storefront and add some products to the shopping cart.
4. Go to the admin and delete the customer group created on step 1.
5. Return to the storefront and proceed to checkout.
* Expected result: a customer should be able to complete an order without any issue.
* Actual result: an error "No such entity id = ... " displayed. Shipping methods are not rendered.
![missin_customer_group](https://user-images.githubusercontent.com/12259690/85865520-b1104180-b7ce-11ea-908a-8c32c8db2d44.png)

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#29327: Fixing bug with non-existent customer group breaking quote
